src/app/workflow/versions/versions.component.ts

```diff
@@ -148,20 +192,41 @@ export class VersionsWorkflowComponent extends Versions implements OnInit, OnCha
       if (workflow) {
         this.defaultVersion = workflow.defaultVersion;
       }
-      this.dtOptions = {
-        bFilter: false,
-        bPaginate: false,
-        columnDefs: [
-          {
-            orderable: false,
-            targets: this.setNoOrderCols(),
-          },
-        ],
-      };
       this.publicPageSubscription();
     });
+    this.dataSource = new VersionsDataSource(this.workflowsService);
+    this.versionsLength$ = this.dataSource.versionsLengthSubject$;
+  }
+
+  loadVersions(publicPage: boolean) {
+    let direction: 'asc' | 'desc';
+    switch (this.sort.direction) {
+      case 'asc': {
+        direction = 'asc';
+        break;
+      }
+      case 'desc': {
+        direction = 'desc';
+        break;
+      }
+      default: {
+        direction = 'desc';
+      }
+    }
+    this.dataSource.loadVersions(
+      publicPage,
+      this.workflowId,
+      direction,
+      this.paginator.pageIndex * this.paginator.pageSize,
+      this.paginator.pageSize,
+      this.sortCol
+    );
   }
 
+  ngOnDestroy() {
```

Comment:
I don't think you need this method. There's already an `ngOnDestroy` implementation in entry-tab.ts that's getting duplicated here.